crossrefindexer/gzip.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"compress/gzip"
+	"encoding/json"
+	"io"
+	"log"
+	"os"
+)
+
+// GzipReader creates a new Reader of a gziped file.
+//
+// It is the caller's responsibility to call Close on the Reader when done.
+func GzipReader(r io.Reader) (*gzip.Reader, error) {
+	gzipReader, err := gzip.NewReader(r)
+	if err != nil {
+		return nil, err
```

Comment:
I think you either should add some context to the error describing what went wrong by `return nil, errors.Wrap(err, "some description")` or you can simplify to just `return gzip.NewReader(r)`. 

Personally I would write:
```go
gzipReader, err := gzip.NewReader(r)
return gzipReader, errors.Wrap(err, "create gzip reader`)
```